lib/bots/versionbot.ts

```diff
@@ -212,11 +212,38 @@ export class VersionBot extends GithubBot.GithubBot {
             // Go through all the commits. We're looking for, at a minimum, a 'change-type:' tag.
             for (let commit of commits) {
                 const commitMessage: string = commit.commit.message;
-                const invalidCommit = !commitMessage.match(/^change-type:\s*(patch|minor|major)\s*$/mi);
 
-                if (!invalidCommit) {
-                    changetypeFound = true;
-                    break;
+                // Split the commits up into lines, and find the last line with any whitespace in.
+                // Whilst we tend to ask for:
+                //  <header>
+                //
+                //  <body>
+                //
+                //  <footer>
+                // This code will actually let you get away with:
+                //  <header>
+                //
+                //  <footer>
+                // As sometimes a development patch may be self-explanatory in the header alone.
+                const lines = commitMessage.split('\n');
+                let index = 0;
+                for (index = lines.length - 1; index >= 0; index -= 1) {
+                    if (lines[index].match(/$\s*^/)) {
```

Comment:
This is interesting. Given this is true, this PR won't work at all - `index` will always be 0 after this loop, so we'll never find a change-type in any commit, and this PR will entirely stop the bot in its tracks.

...Can I use this to nudge us towards unit & long-term maybe even integration testing for this? With core infrastructure like this, it'd be nice if it was really hard to totally break it without realising..